CORE-17355 - Reduce number of p2p e2e sessions maintained per pair of vnodes

nikinagy opened the pull request:
In order to increase our performance, this PR reduces number of p2p e2e sessions maintained per pair of vnodes:
- for member <-> MGM communication reduce to 1
- for member <-> member communication reduce to 1

nikinagy committed:
add new config values

nikinagy committed:
bump api version

nikinagy committed:
remove owners

nikinagy committed:
Merge branch 'release/os/5.1' into nikinagy/performance-enhancement

nikinagy committed:
Merge remote-tracking branch 'origin/nikinagy/performance-enhancement' into nikinagy/CORE-17355

nikinagy committed:
remove owners and rename api revision

nikinagy committed:
Merge branch 'feature/large-network-improvements' into nikinagy/CORE-17355

# Conflicts:
#	gradle.properties

nikinagy committed:
rename api revision

yift-r3 commented on data/config-schema/src/main/resources/net/corda/schema/configuration/p2p.linkManager/1.0/corda.p2p.linkManager.json:
```diff
@@ -89,10 +89,16 @@
     },
-    "sessionsPerPeer": {
-      "description": "The number of actively maintained sessions between two peers.",
+    "sessionsPerPeerForMgm": {
```
Would we have backward compatibility issues?

Should/can we keep the `sessionsPerPeer` as a deprecated configuration?

yift-r3 replied:
Also, maybe it can be less flat? Something like:
```
"sessionsPerPeer": {
  "forMgm": 1,
  "forMember": 2
}
```

nikinagy replied:
Good point!

charlieR3 committed:
CORE-0 Set up network team 5.2 branch

nikinagy committed:
address comments

nikinagy committed:
Merge remote-tracking branch 'origin/feature/network_5_2' into nikinagy/CORE-17355

# Conflicts:
#	gradle.properties

nikinagy committed:
revert change

nikinagy committed:
revert change

charlieR3 committed:
CORE-18229 Set code owners on feature branch (#1329)

Set the code owners on the platform network branch to just the platform network team. Not to be merged to the 5.2 branch once it is created.

nikinagy committed:
Merge remote-tracking branch 'origin/feature/network_5_2' into nikinagy/CORE-17355

# Conflicts:
#	.github/CODEOWNERS

yift-r3 reviewed (dismissed):
LGTM. Should we have someone from docs/product look at the descriptions?

nikinagy committed:
fix naming

nadinequinr3 commented on data/config-schema/src/main/resources/net/corda/schema/configuration/p2p.linkManager/1.0/corda.p2p.linkManager.json:
```diff
@@ -91,10 +91,29 @@
+      "default": {},
+      "properties": {
+        "forMgm": {
+          "description": "The number of actively maintained sessions between two peers if one of them is an MGM.",
```
Might be nice to give a little more context. For example: 
```suggestion
          "description": "The number of actively maintained sessions between two peers if one of them is an MGM. Communication between an MGM and a member is not considered critical and so the default value of 1 is recommended to improve performance.",
```

nikinagy replied:
addressed in [980cb7f](https://github.com/corda/corda-api/pull/1307/commits/980cb7f1e2b40256508cee92a25f223b288ffb15)

nadinequinr3 commented on data/config-schema/src/main/resources/net/corda/schema/configuration/p2p.linkManager/1.0/corda.p2p.linkManager.json:
```diff
@@ -91,10 +91,29 @@
+          "minimum": 1
+        },
+        "forMembers": {
+          "description": "The number of actively maintained sessions between two peers if both peers are members.",
```
Might be nice to give a little more context. For example:

```suggestion
          "description": "The number of actively maintained sessions between two peers if both peers are members. The default value of 2 is recommended because the chance of both sessions failing on the same link manager is relatively low. If high-availability is critical, R3 recommended deploying multiple link manager instances.",
```

nikinagy replied:
addressed in [980cb7f](https://github.com/corda/corda-api/pull/1307/commits/980cb7f1e2b40256508cee92a25f223b288ffb15)

nikinagy committed:
address comments

charlieR3 committed:
Merge branch 'release/os/5.1' into charlie/network_5_2-merge-nov-10

charlieR3 committed:
Merge pull request #1333 from corda/charlie/network_5_2-merge-nov-10

CORE-18278 Merge the release 5.1 branch into the network team 5.2 feature branch

nikinagy committed:
Merge remote-tracking branch 'origin/feature/network_5_2' into nikinagy/CORE-17355

nikinagy committed:
Merge remote-tracking branch 'origin/release/os/5.2' into nikinagy/CORE-17355

# Conflicts:
#	.github/CODEOWNERS
#	gradle.properties

nikinagy committed:
Merge remote-tracking branch 'origin/release/os/5.2' into nikinagy/CORE-17355

williamvigorr3 commented on data/config-schema/src/main/resources/net/corda/schema/configuration/p2p.linkManager/1.0/corda.p2p.linkManager.json:
```diff
@@ -91,10 +91,29 @@
-      "type": "integer",
-      "default": 4,
+      "type": ["integer", "null"],
+      "default": null,
```
I think having two config fields `sessionsPerPeer` and `numOfSessionsPerPeer` with the same description is confusing, for a user. Is `sessionsPerPeer` deprecated? I know I am coming to this quite late but I am trying to do the same thing for `CORE-17960`.

nikinagy replied:
You are right I should have marked it as "deprecated field"

nikinagy committed:
address comments
